### ISSUE TITLE
#include handling broken in Shader Generation

### ISSUE THREAD

**Issue by ld-kerley** [AcademySoftwareFoundation/MaterialX#2097]
While helping Masuo get the hex tiling PR working in MSL, I spent a long time being confused....

Commenting out `#include` lines does not stop them from being included.

```
//#include "mx_math.glsl"
```

The `mx_math.glsl` file was still being included, causing compile errors.

In this case we don't actually need to include this file, but in general we need shader generation to be robust, as its a bit of a blackbox for shader developers.

**Comment by jstone-lucasfilm** [comment]
Good catch, @ld-kerley and @msuzuki-nvidia, and I believe that logic is currently implemented here in the shader generation system:

https://github.com/AcademySoftwareFoundation/MaterialX/blob/ed450e82de41a9a5d4273c49babfdb37b18b922e/source/MaterialXGenShader/ShaderStage.cpp#L315
